flask_server.py

```diff
@@ -24,35 +24,34 @@ def get_log_file_list():
 def create_markup(base64_data, media_type = None, username = None):
     if username != None:
         return Markup((f"<hr><div class=\"username-text\">{username}</div><br>"))
-    if media_type in ["video/mp4", "img/png"]:
-        content_tag = "img" if media_type == "img/png" else "video controls"
-        return Markup(f"<{content_tag} src=\"data:{media_type};base64,{base64_data}\" class=\"rendered-stories\"></{content_tag}>")
-
-def render_base64_media(base64_media):
-    rendered_base64_media = []
+        
+def get_instance_media(base64_media):
+    instance_media = []                      # instance_media struct:  {"type": username, "data": base64_data or username}          base64data = {"content_tag", "media_type", "data"}
     last_username = None
     for base64_data, media_type, username in base64_media:
+        content_tag = "img" if media_type == "img/png" else "video controls"
         if username != last_username:
-            rendered_base64_media.append(create_markup(base64_data, username = username))
+            instance_media.append({"type": "username", "data": username})
             last_username = username
-        rendered_base64_media.append(create_markup(base64_data, media_type))
-    return rendered_base64_media
+        instance_media.append({"type": "media", "data": {"content_tag": content_tag,"media_type": media_type,"base64_data": base64_data}})     
+    return instance_media
 
 def get_folders(path, url):
-    rendered_folders = [] # List of {url: X name: Y}
+    rendered_folders = [] # List of {url: X, name: Y}
     if not os.path.exists(path) : return []
     for folder in os.listdir(path):
         rendered_folders.append({'type': 'folder', 'url': f"{url}{folder}", 'name': f"{folder}"})
     return rendered_folders
 
-def get_rendered_media(path):
+def get_media(path):
     rendered_media = []
```

Comment:
It's not rendered media, as it still a json and not an html string. 

Consider changing the name 

to_render_media?